src/NuGet.Core/NuGet.Packaging/PackageReaderBase.cs

```diff
@@ -527,7 +527,13 @@ protected static string GetNuspecFile(IEnumerable<string> files)
 
             if (nuspecPaths.Count == 0)
             {
-                throw new PackagingException(Strings.MissingNuspec);
+                // Find the package directory name by inspecting first file in the folder
+                FileInfo file = new FileInfo(files.ElementAtOrDefault(0));
```

Comment:
what if `files` is empty? will `new FileInfo(null)` throw? If so, I'd just use `new FileInfo(files.First())` as that tells anyone reading the code that `files` must have at least one string.